Update Go to 1.23

tklauser opened the pull request:
Cherry-picked from https://github.com/cilium/cilium/pull/34362

It seems the bot is unable to update `cilium-{builder,runtime}` images on that PR (https://github.com/cilium/cilium/actions/runs/10523063087/job/29157020472?pr=34362) so doing it manually on this PR.

Closes #34362

aanm reviewed (approved):
@tklauser it looks the issue is because `go.mod` has 1.23 in it. I've reverted that change alone.

kaworu reviewed (approved):
Hubble changes LGTM

sayboras reviewed (approved):
LGTM

Just curious if we need to remove dependabot config for go directive in go.mod, or just let the upgrade fail :thinking:

tklauser committed:
test/k8s: always unmap v4-in-v6 addresses

Parsing the address from curl output can lead to abiguity wrt. v4-in-v6
addresses. Make sure to unwrap them and only care about either IPv4 or
IPv6 addresses.

Fixes: 159be8f47ea1 ("Updates k8sTest pkg to use netip.Addr")

Signed-off-by: Tobias Klauser <tobias@cilium.io>

sayboras commented on go.mod:
```diff
@@ -1,7 +1,7 @@
 module github.com/cilium/cilium
 
 // renovate: datasource=golang-version depName=go
-go 1.22.0
```
should we revert this change as mentioned in above comment ?

cilium-renovate[bot] committed:
chore(deps): update go to v1.23.0

Signed-off-by: cilium-renovate[bot] <134692979+cilium-renovate[bot]@users.noreply.github.com>

sayboras committed:
images: update cilium-{runtime,builder}

Signed-off-by: Cilium Imagebot <noreply@cilium.io>
Signed-off-by: Tam Mach <tam.mach@cilium.io>
